### ISSUE TITLE
Migration Guide for LPS Breaking Changes

### ISSUE THREAD

**Issue by benjamincharity** [GetTerminus/terminus-oss#210]
Several components have had breaking changes to support the LPS project. Breaking changes and migration notes live in the `CHANGELOG.md` in the respective package.

Below is a collection of the breaking change migrations:

# Migration Notes

To update packages (including breaking changes): 

```bash
# Update all Terminus packages to the latest version allowed by existing version ranges:
$ yarn upgrade --scope @terminus

# Update all Terminus packages to the latest version INCLUDING breaking changes:
$ yarn upgrade --latest --scope @terminus
```

### Theme

If passing in a theme, use one of the new button themes: `default|secondary|warning`:

```diff
-<ts-button theme="primary">Click Me!</ts-button>
+<ts-button theme="default">Click Me!</ts-button>

-<ts-menu theme="accent">Select Item</ts-menu>
+<ts-menu theme="secondary">Select Item</ts-menu>

-<ts-file-upload theme="warn"></ts-file-upload>
+<ts-file-upload theme="warning"></ts-file-upload>
```

See `TsButtonThemeTypes` for Types and `tsButtonThemes` for an array of allowed themes.

### Hollow

If using `format="hollow"` change to `theme="secondary"`:

```diff
-<ts-button format="hollow">Click Me!</ts-button>
+<ts-button theme="secondary">Click Me!</ts-button>

-<ts-menu format="hollow">Select Item</ts-menu>
+<ts-menu theme="secondary">Select Item</ts-menu>
```

## Icon/IconButton/Button

If passing in an icon, pass in a FontAwesome icon reference instead of a string name:

```typescript
import { faHome } from '@fortawesome/pro-solid-svg-icons/faHome';
...
public home = faHome;
```

```diff
-<ts-button iconName="home">Go Home</ts-button>
+<ts-button [icon]="home">Go Home</ts-button>

-<ts-icon>home</ts-icon>
+<ts-icon [icon]="home"></ts-icon>
```

> NOTE: Multiple versions of each icon exist. We should always import from the 'solid' collection.
> NOTE: Even free icons can be imported from the 'pro' package: `@fortawesome/pro-solid-svg-icons/...`

### Installing the Pro FontAwesome version

See full instructions here: https://fontawesome.com/how-to-use/on-the-web/setup/using-package-managers#installing-pro

TL;DR

Add an `.npmrc` file with these contents to your project:

```
@fortawesome:registry=https://npm.fontawesome.com/
//npm.fontawesome.com/:_authToken=${FONTAWESOME_NPM_AUTH_TOKEN}
```

Add an environment variable (local and in your CI environment) called `FONTAWESOME_NPM_AUTH_TOKEN`. The token value can be found in the UXE, Design or Valor 1Password vaults.

## Button

Long-time deprecated misspelled format `collapsable` has been fully removed in favor of `collapsible`.

## Checkbox/Link

No longer accepts the `theme` input:

```diff
-<ts-checkbox theme="primary">My checkbox</ts-checkbox>
+<ts-checkbox>My checkbox</ts-checkbox>

-<ts-link theme="primary">My link</ts-link>
+<ts-link>My link</ts-link>
```

## CSS Resources

Remove any imports or `<link>` tags importing Material Icons:

```diff
-<link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
```

Update the imported font families:

```diff
-<link href="https://fonts.googleapis.com/css?family=Roboto:400,500,700" rel="stylesheet">
+<link href="https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,400;0,500;0,700;1,400&display=swap" rel="stylesheet">
```

# Links

- https://github.com/GetTerminus/terminus-oss/blob/release/libs/ui/button/CHANGELOG.md#200-migration-notes
- https://github.com/GetTerminus/terminus-oss/blob/release/libs/ui/checkbox/CHANGELOG.md#200-migration-notes
- https://github.com/GetTerminus/terminus-oss/blob/release/libs/ui/icon/CHANGELOG.md#200-migration-notes
- https://github.com/GetTerminus/terminus-oss/blob/release/libs/ui/icon-button/CHANGELOG.md#200-migration-notes
- https://github.com/GetTerminus/terminus-oss/blob/release/libs/ui/file-upload/CHANGELOG.md#200-migration-notes
- https://github.com/GetTerminus/terminus-oss/blob/release/libs/ui/link/CHANGELOG.md#200-migration-notes
- https://github.com/GetTerminus/terminus-oss/blob/release/libs/ui/menu/CHANGELOG.md#200-migration-notes

---

### EDITS/ADDITIONS

- Link no longer accepts `theme` input
- Fixed incorrect GitHub autocomplete `@terminus-devops` to `@terminus`
